openssl/src/ocsp.rs

```diff
@@ -14,6 +15,27 @@ use crate::x509::{X509Ref, X509};
 use crate::{cvt, cvt_p};
 use openssl_macros::corresponds;
 
+// Sentinel value used when next_update is not present in OCSP response
+// This represents the maximum possible time (9999-12-31 23:59:59 UTC)
+static SENTINEL_MAX_TIME: OnceLock<Asn1Time> = OnceLock::new();
+
+fn get_sentinel_max_time() -> &'static Asn1GeneralizedTimeRef {
+    // ASN1_TIME and ASN1_GENERALIZEDTIME have compatible representations
+    // and can be safely cast. ASN1_TIME is the generic type that can hold
+    // both UTCTIME and GENERALIZEDTIME. For dates outside 1950-2049,
+    // it automatically uses GENERALIZEDTIME format.
```

Comment:
This is confusingly and misleadingly describing the RFC 5280 semantics.

In OpenSSL the `ASN1_TIME_set_string()` function (underlying `Asn1Time::from_str()`) does *not* use these semantics, that's what `ASN1_TIME_set_string_X509()` is for. It decides between UtcTime and GeneralizedTime solely on basis of the length of the string. It will happily accept `20251106080730Z` as a GeneralizedTime.

Reply:
`nextUpdate` is _always_ a GeneralizedTime -- it's not a Time (CHOICE) like many other X.509 things.